kinto_http/tests/test_session.py

```diff
@@ -91,7 +93,39 @@ def test_passed_data_is_encoded_to_json(self):
                         data={'foo': 'bar'})
         self.requests_mock.request.assert_called_with(
             'post', 'https://example.org/test',
-            json={"data": {'foo': 'bar'}}, headers=self.requests_mock.request.return_value.headers)
+            data='{"data": {"foo": "bar"}}',
+            headers=self.requests_mock.request.post_json_headers)
+
+    def test_passed_datetime_data_is_encoded_to_json(self):
+        response = fake_response(200)
+        self.requests_mock.request.return_value = response
+        session = Session('https://example.org')
+        session.request('post', '/test',
+                        data={'foo': datetime(2018, 6, 22, 18, 00)})
+        self.requests_mock.request.assert_called_with(
+            'post', 'https://example.org/test',
+            data='{"data": {"foo": "2018-06-22T18:00:00"}}',
+            headers=self.requests_mock.request.post_json_headers)
+
+    def test_passed_random_python_data_fails_to_be_encoded_to_json(self):
+        response = fake_response(200)
+        self.requests_mock.request.return_value = response
+        session = Session('https://example.org')
+        with pytest.raises(TypeError) as exc:
+            session.request('post', '/test',
+                            data={'foo': object()})
+        assert str(exc.value) == "Type <class 'object'> is not serializable"
+
+    def test_passed_date_data_is_encoded_to_json(self):
+        response = fake_response(200)
+        self.requests_mock.request.return_value = response
+        session = Session('https://example.org')
+        session.request('post', '/test',
+                        data={'foo': date(2018, 6, 22)})
+        self.requests_mock.request.assert_called_with(
+            'post', 'https://example.org/test',
+            data='{"data": {"foo": "2018-06-22"}}',
+            headers=self.requests_mock.request.post_json_headers)
```

Comment:
nit: These 3 tests share a common `setUp`. Do you think you could put them in a dedicated test suite?